src/annotations/api/index.ts

```diff
@@ -0,0 +1,34 @@
+// Types
+import {PostAnnotationResponse, PostAnnotationPayload} from 'src/types'
+
+// Libraries
+import axios, {AxiosResponse} from 'axios'
+
+// Constants
+import {API_BASE_PATH} from 'src/shared/constants'
+
+// URL
+const url = `${API_BASE_PATH}api/v2private/annotations`
+
+export const writeAnnotation = async (
+  annotations: PostAnnotationPayload[]
+): Promise<AxiosResponse<PostAnnotationResponse[]>> => {
+  const res = await axios.post(url, annotations)
+
+  if (res.status >= 300) {
+    throw new Error(res.data?.message)
+  }
+
+  const transformedResponse = res.data.map(d => {
+    const {'start-time': startTime, 'end-time': endTime, ...props} = d
+    const newData = {
+      startTime,
+      endTime,
+      ...props,
+    }
```

Comment:
this is fine. the thing is, we know what the structure of `PostAnnotationResponse` is, so we can just create an object by hand:

```typescript
const newData = {
  startTime: d['start-time'],
  endTime: d['end-time'],
  stream: d.stream,
  summary: d.summary,
  stickers: d.stickers
}
```

this is explicit. it lays out in clear, unambiguous detail, what properties from the api are being converted to properties of our domain. this will show up when people search through the codebase for either of these types of variables (e.g. searching for `'startTime'` or `'start-time'`). it clearly documents what is happening in local context without making someone look at the definition of `PostAnnotationResponse` to see what properties are being splatted out in `...props`

we want to optimize code to be dead simple and extremely clear and unambiguous when people read it. sometimes that means adding more lines to an existing piece of code. that is okay, if it's in service of comprehension.